testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -885,6 +885,10 @@ def test_segmentgroup_repr(self, u):
     def test_segmentgroup_str(self, u):
         sg = u.segments[:10]
         assert str(sg) == '<SegmentGroup [<Segment 4AKE>]>'
+    
+    def test_atomgroupresidue_dtype(self,u):
+        rg = list(u.atoms[100:130].residues)
+        assert rg == '[<Residue LEU, 6>, <Residue GLY, 7>, <Residue ALA, 8>]'
```

Comment:
This will fail. `rg` is a list of residue objects and not strings.

Try
```suggestion
assert len(rg) == 3
```
or something like that.